package-best-practices.md

```diff
@@ -0,0 +1,41 @@
+# Best practices when submitting a package for public use
+
+Packages are a key part of the dbt experience. Practitioners and vendors alike can [contribute to the knowledge loop](https://github.com/dbt-labs/corp/blob/main/values.md#we-contribute-to-the-knowledge-loop) by enabling compelling, batteries-included use cases.
+
+There are hundreds of packages on the dbt Package Hub already, and over time we have identified a handful of common mistakes that new package authors make as they take a self-contained package and prepare it for publication. We have listed these below to help you and your users get the best experience.
+
+Although the dbt Labs team completes a cursory review of new packages before adding them to the Package Hub, ultimately you are responsible for your package's performance.
+
+To avoid ambiguity, the bullets below are pretty formal, but we are a friendly bunch! If you need help preparing a package for deployment to the dbt Package Hub, reach out in [#package-ecosystem](https://getdbt.slack.com/archives/CU4MRJ7QB/) in the [dbt Community Slack](https://getdbt.com/community).
+
+---
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+### First run experience
+- Packages SHOULD contain a README file which explains how to get started with the package and customise its behaviour.
+
+### Customisability 
+- Packages MUST NOT hard-code table references, and MUST use `ref` or `source` instead.
+- Packages MAY provide a mechanism (such as variables) to enable/disable certain subsets of functionality.
+- Packages for data transformation:
+    - SHOULD provide a mechanism (such as variables) to customise the location of source tables.
+    - SHOULD NOT assume database/schema names in sources. 
+    - MAY assume table names, particularly if the package was built to support tables created by a known tool.
+
+### Dependencies
+- Packages requiring a minimum version of `dbt-core` MUST declare it in the `require-dbt-version` property of `dbt_project.yml`.
+- Packages requiring a minimum version of `dbt-core` SHOULD declare compatibility with all subsequent minor and patch releases of that major version. 
+    - For example, a package which depends on functionality added in dbt Core 1.2 SHOULD set its `require-dbt-version` property to `[">=1.2.0", "<2.0.0"]`.
+Packages with a dependency on another package:
+    - MAY, if the dependent package is major-version-zero, pin to the current minor version only.
```

Comment:
Does "dependent" make clear that it is the package that the main package is importing? I think it does until I think too hard about dependents in the real world normally being children, which would make them the downstream one as opposed to looking upstream. Not sure if I'm too in my head or not, and don't have a better word.

Reply:
Would "requirement" be a less confusing word?

An additional possibility: the "Dependencies" could be split into two parts:
- compatible versions of dbt Core
- compatible versions of package requirements

The first bullet point references `dbt_project.yml` -- maybe there could be one could referencing `packages.yml` also?

- Each package requirement MUST be declared in `packages.yml`.
- Requirement ranges SHOULD be as wide as compatibility allows.

☝️ might need some further word-smithing